File: Module.java
package dev.mmrl;

import android.webkit.JavascriptInterface;

import com.dergoogler.mmrl.platform.file.SuFile;
import com.dergoogler.mmrl.platform.model.ModId;
import com.dergoogler.mmrl.webui.interfaces.WXInterface;
import com.dergoogler.mmrl.webui.interfaces.WXOptions;

import org.jetbrains.annotations.NotNull;

public class Module extends WXInterface {
    private final ModId modId;
    private final SuFile webrootDir;
    private final SuFile moduleDir;
    private final SuFile moduleConfigDir;
    private final SuFile systemDir;


    public Module(WXOptions wxOptions) {
        super(wxOptions);

        this.modId = this.getModId();
        this.webrootDir = ModId.Companion.getWebrootDir(modId);
        this.moduleDir = ModId.Companion.getModuleDir(modId);
        this.moduleConfigDir = ModId.Companion.getModuleConfigDir(modId);
        this.systemDir = ModId.Companion.getSystemDir(modId);
    }

    @NotNull
    @Override
    public String getName() {
        return "module";
    }

    @JavascriptInterface
    public String getId() {
        return modId.toString();
    }

    @JavascriptInterface
    public String getAdbDir() {
        return ModId.Companion.getAdbDir(modId).getPath();
    }

    @JavascriptInterface
    public String getConfigDir() {
        return ModId.Companion.getConfigDir(modId).getPath();
    }

    @JavascriptInterface
    public String getModuleConfigDir() {
        return moduleConfigDir.getPath();
    }

    @JavascriptInterface
    public String getModuleConfigDir(String... paths) {
        SuFile file = new SuFile(moduleConfigDir, paths);
        return file.getPath();
    }

    @JavascriptInterface
    public String getModulesDir() {
        return ModId.Companion.getModulesDir(modId).getPath();
    }

    @JavascriptInterface
    public String getModuleDir() {
        return moduleDir.getPath();
    }

    @JavascriptInterface
    public String getModuleDir(String... paths) {
        SuFile file = new SuFile(moduleDir, paths);
        return file.getPath();
    }

    @JavascriptInterface
    public String getWebRootDir() {
        return webrootDir.getPath();
    }

    @JavascriptInterface
    public String getWebRootDir(String... paths) {
        SuFile file = new SuFile(webrootDir, paths);
        return file.getPath();
    }

    @JavascriptInterface
    public String getSystemDir() {
        return systemDir.getPath();
    }

    @JavascriptInterface
    public String getSystemDir(String... paths) {
        SuFile file = new SuFile(systemDir, paths);
        return file.getPath();
    }

    @JavascriptInterface
    public String getPropFile() {
        return ModId.Companion.getAdbDir(modId).getPath();
    }
}

File: plugin/src/main/kotlin/dev/mmrl/module/Module.kt
@file:Suppress("unused")

package dev.mmrl.module

import android.webkit.JavascriptInterface
import com.dergoogler.mmrl.platform.model.ModId.Companion.adbDir
import com.dergoogler.mmrl.platform.model.ModId.Companion.configDir
import com.dergoogler.mmrl.platform.model.ModId.Companion.moduleConfigDir
import com.dergoogler.mmrl.platform.model.ModId.Companion.moduleDir
import com.dergoogler.mmrl.platform.model.ModId.Companion.modulesDir
import com.dergoogler.mmrl.platform.model.ModId.Companion.propFile
import com.dergoogler.mmrl.platform.model.ModId.Companion.systemDir
import com.dergoogler.mmrl.platform.model.ModId.Companion.webrootDir
import com.dergoogler.mmrl.webui.interfaces.WXInterface
import com.dergoogler.mmrl.webui.interfaces.WXOptions
import dev.mmrl.internal.WXUInterface

class Module(wxOptions: WXOptions) : WXUInterface(wxOptions) {
    override var name = "module"

    @get:JavascriptInterface
    val id: String = modId.toString()

    @get:JavascriptInterface
    val adbDir: String = modId.adbDir.path

    @get:JavascriptInterface
    val configDir: String = modId.configDir.path

    @get:JavascriptInterface
    val moduleConfigDir: String = modId.moduleConfigDir.path

    @get:JavascriptInterface
    val modulesDir: String = modId.modulesDir.path

    @get:JavascriptInterface
    val moduleDir: String = modId.moduleDir.path

    @get:JavascriptInterface
    val webRootDir: String = modId.webrootDir.path

    @get:JavascriptInterface
    val systemDir: String = modId.systemDir.path

    @get:JavascriptInterface
    val propFile: String = modId.propFile.path
}
